lib/Support/Check.cpp

```diff
@@ -0,0 +1,123 @@
+//===- Check.cpp - Check test results  ------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+//
+//===----------------------------------------------------------------------===//
+
+#include "Support/Check.h"
+
+inline bool isdenorm(float f) {
+  return (std::numeric_limits<float>::denorm_min() <= f &&
+          f < std::numeric_limits<float>::min()) ||
+         (-std::numeric_limits<float>::min() < f &&
+          f <= -std::numeric_limits<float>::denorm_min());
+}
+
+bool compareFloatULP(
+    const float &fsrc, const float &fref, unsigned ULPTolerance,
```

Comment:
Please follow LLVM conventions for naming of things across this file:
https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly